peps/pep-806.rst

```diff
@@ -0,0 +1,292 @@
+and potential for errors - it's easy to violate properties that syntactic use
+of context managers would guarantee, such as 'last-in, first-out' order.
+
+
+Workaround: ``AsyncExitStack``-based helper
+-------------------------------------------
+
+We could also implement a ``multicontext()`` wrapper, which avoids some of the
+downsides of direct use of :class:`~contextlib.AsyncExitStack`:
+
+.. code-block:: python
+
+    async with multicontext(
+        acquire_lock(),
+        open('file'),
+    ) as (f, _):
+        ...
+
+However, this helper breaks the locality of ``as`` clauses, which makes it
+easy to accidentally mis-assign the yielded variables (as in the code sample).
+It also requires either distinguishing sync from async context managers using
+something like a tagged union - perhaps overload an operator so that, e.g.,
+``async_ @ acquire_lock()`` works - or else guessing what to do with objects
+that implement both sync and async context-manager protocols.
+Finally, it has the error-prone semantics around exception handling which led
+`contextlib.nested()`__ to be deprecated in favor of the multi-argument
+``with`` statement.
+
+__ https://docs.python.org/2.7/library/contextlib.html#contextlib.nested
+
+
+Syntax: allow ``async with sync_cm, async_cm:``
+-----------------------------------------------
+
+An early draft of this proposal used ``async with`` for the entire statement
+when mixing context managers, *if* there is at least one async context manager:
+
+.. code-block:: python
+
+    # Rejected approach
+    async with (
+        acquire_lock(),
+        open('config.json') as f,  # actually sync, surprise!
+    ):
+        ...
+
+Requiring an async context manager maintains the syntax/scheduler link, but at
+the cost of setting invisible constraints on future code changes.  Removing
+one of several context managers could cause runtime errors, if that happened
+to be the last async context manager!
+
+Explicit is better than implicit.
+
+
+Syntax: ban single-line ``with async ...``
+------------------------------------------
+
+Our proposed syntax could be restricted, e.g. to place ``async`` only as the
+first token of lines in a parenthesised multi-context ``with`` statement.
+This is indeed how we recommend it should be used, and we expect that most
+uses will indeed follow this pattern.
+
+While an option to write either ``async with ctx():`` or ``with async ctx():``
+may cause some small confusion due to ambiguity, we think that enforcing a
+preferred style via the syntax would make Python more confusing to learn,
+and thus prefer simple syntactic rules plus community conventions on how to
+use them.
+
+
+Acknowledgements
+================
+
+Thanks to Rob Rolls for `proposing`__ ``with async``.  Thanks also to the many
+other people with whom we discussed this problem and possible solutions at the
+PyCon 2025 sprints, on Discuss, and at work.
```

Comment:
You could choose to list your place of work, should you so wish.
```suggestion
PyCon 2025 sprints, on Discourse, and at work.
```

Reply:
I'm open about that on my personal website, but heard enough negativity about it at PyCon this year that I feel it would on net distract from this PEP 😞

Reply:
I've just looked at your website -- looks like a very cool job (& a very interesting firm!), but I understand the hesitation. 

Sorry for  not knowing previously, I'm not the best at keeping track of these things!